### ISSUE TITLE
Unable to add hoobs to homekit

### ISSUE THREAD

**Issue by jkhoob** [hoobs-org/HOOBS#1398]
I Just installed hoobs v4.0.87 on raspberry pi and I was able to connect plugins and accessories to hoobs. However, when I try to scan QR code for any bridges, I'm getting "Accessory Not Found". 

In the past, I remember seeing default QR code to connect hoobs to homekit and accessories would get added automatically once you connect. It appears this has changed and requires to add individual bridges per accessories. 

I'm not sure if I'm missing a step, I was not able to find anything on the web. All other instructions was based on previous version. 

Hoobs is on the same network to every device in my house. I have tried restart, change username & pin and bridge advertiser. Nothing appears to work.

**Comment by mkellsy** [comment]
Is your HOOBS device and your iPhone or iPad on the same network. Is your hardline and Wi-Fi on different networks?
You may have to add a HomePod or Apple TV to the same network as your HOOBS device so Apple Home can relay HomeKit to your other devices.

**Comment by jkhoob** [comment]
Yes, everything is on the same network, subnet, vlan, etc. I was able to connect plugins to devices wyze cam via RTSP FFmpeg, tplink smarthome, myQ, etc. However, whenever I try to add them to homekit, I'm getting accessories not found. However, I'm running 5ghz on my iphone and apple tv but hoobs Pi is on ethernet. Does this matter?

**Comment by mkellsy** [comment]
OK. Check to see if a bridge with the same name is defined in HomeKit.
Home Settings -> Hubs & Bridges
You may have to remove the bridge from this list before pairing.

**Comment by jkhoob** [comment]
Under hubs & Bridges I have Apple-TV and Eufy homebase. That is all.

**Comment by mkellsy** [comment]
The last thing you can try is to clear the bridge cache.
Bridges Screen -> Select the bridge in question then click the Cache button
There is a button on the bottom called Clear Cache, click it.

This will regenerate the UUID for the bridge and then you should be able to add it.

If this doesn't work, you may need to remove the bridge and reinstall the plugins to a different bridge.

**Comment by jkhoob** [comment]
None of those worked... what do you mean by reinstall to different bridge? Create custom bridge? I'm not sure if I'm doing this wrong but I have created bridge per plugin. Am I supposed to add all the plugins to single bridge?

**Comment by jkhoob** [comment]
Am I the only one experiencing this issue? I have factory reset hoobs and add just one bridge and plugin, it is working fine from hoobs. As soon as I connect hoobs bridge to homekit, I'm getting accessory not found. Upgraded apple tv, iphone and hoobs. Not sure what I'm missing...

**Comment by mkellsy** [comment]
Yes you are currently.
Making a new bridge is like resetting. I am concerned that a fresh build isn’t going to help.

check your up address on your iPhone. Settings -> Wi-Fi then tap the i on the connected Wi-Fi. Note the IP address and Router.

Then on HOOBS go to the terminal and run

ip address

compare the IP address and router. They should all have the same 192.168.x.x (a.b.c)

The only other thing I can think of is your iPhone might need to be upgraded.

**Comment by jkhoob** [comment]
Yes, it is same network, VLAN, subnet. Only difference, hoob is on ethernet and rest are WiFi. Is there way to enable raspberry pi WiFi on hoobs?

**Comment by mkellsy** [comment]
https://support.hoobs.org/docs/60e0b978646faaa152f10dcc

**Comment by just808b** [comment]
Did you solve the problem? I can’t add anything that requires configuration

**Comment by boomvalt** [comment]
I'm getting a related problem it seems. Whereas I had no problem with homebridge (apart from its frontend) I get no homekit connection QR on the HOOBS Dashboard and no documentation on how to get one.

**Comment by mkellsy** [comment]
The QR codes are per bridge. Go to the Bridges screen and select the bridge you want to pair.

**Comment by jkhoob** [comment]
Still having the same issue, I even tried connecting with new homekit under my wife's acct and still no good. I have latest code and there is no documentation. The main difference appears to be sharing a bridge QR code vs creating individual bridge per plugins. I have no issues setting up plugins to hoobs but I'm unable to add bridges to homekit.

**Comment by mkellsy** [comment]
The QR codes are not shared. They are different per bridge. We do have documentation on this here https://support.hoobs.org/docs/60e0c685646faaa152f10dd8

We also have documentation on switching to WiFi here https://support.hoobs.org/docs/60e0b978646faaa152f10dcc

**Comment by nullbreached** [comment]
I’m having the same issue. Everything hoobs, seems up and running but I cannot add the hoobs bridge via the QR code.

**Comment by nullbreached** [comment]
Here’s my IP address info via the command “ip address” via hooks terminal. (I’m using an iPad at the moment and the terminal function in a web browser won’t allow me to select text.)
![222D3343-B9DF-489C-B096-0BF58B439F0B](https://user-images.githubusercontent.com/9218674/130360463-47c5d65b-0bb4-4688-a4d3-23d7c39f2293.jpeg)

**Comment by mkellsy** [comment]
Can you get the pi address and subnet from your iPad?

**Comment by nullbreached** [comment]
So I finally got stuff connected. But as the issue dragged out I started grasping at more things to try and have no idea what may have worked. But I do think my router has some issues. (Netgear C7000) I did change the DNS servers to Google servers which briefly may have solved the resolving of the hoobs.local domain instead of using the IP address. I think the  router may have problems with mDNS. After that router change, or maybe just a router reboot it also prompted the first time a new message appeared in the process of adding the hub to HomeKit via QR of “this device isn’t a HomeKit approved device …”—which still ended in failing to add the hub but at least it was a change. A subsequent attempt now yielded a “This device is already added” error.  Hopeful, I thought maybe it had worked but the one accessory I had added didn’t appear in HomeKit and the hub also didn’t appear in the list of hubs in the Home app (under Home Settings). Following that I did a full factory reset (via the UI) of Hoobs 4, which didn’t seem to have effect. Then I tried changing the identifier several times, through the UI, no luck. Then I tried changing the name of the hub and the identifier. Suddenly this process worked! TLDR: change the name of the hub and possibly the identifier. I know this likely is entirely unhelpful.

**Comment by mkellsy** [comment]
It sounds like HomeKit tried to connect before, but mDNS (aka Bonjour) may have been blocked by your router. This caused a cached connection. Changing the name and/or identifier is the right thing to do in this case.

I was asking for your iPad IP address so I could compare the addresses from your previous post. I was trying to diagnose your router. If the IP addresses were from the same subnet (mathematically) I was going to suggest looking into your router.

It looks like you figured this out on you own. Good Work ! :)

**Comment by ksegs11** [comment]
> So I finally got stuff connected. But as the issue dragged out I started grasping at more things to try and have no idea what may have worked. But I do think my router has some issues. (Netgear C7000) I did change the DNS servers to Google servers which briefly may have solved the resolving of the hoobs.local domain instead of using the IP address. I think the router may have problems with mDNS. After that router change, or maybe just a router reboot it also prompted the first time a new message appeared in the process of adding the hub to HomeKit via QR of “this device isn’t a HomeKit approved device …”—which still ended in failing to add the hub but at least it was a change. A subsequent attempt now yielded a “This device is already added” error. Hopeful, I thought maybe it had worked but the one accessory I had added didn’t appear in HomeKit and the hub also didn’t appear in the list of hubs in the Home app (under Home Settings). Following that I did a full factory reset (via the UI) of Hoobs 4, which didn’t seem to have effect. Then I tried changing the identifier several times, through the UI, no luck. Then I tried changing the name of the hub and the identifier. Suddenly this process worked! TLDR: change the name of the hub and possibly the identifier. I know this likely is entirely unhelpful.

When you say you changed the name of the hub, are you meaning the bridge? So change the name and the identifier of each individual bridge and then try to add to Apple home?

**Comment by galangel** [comment]
The same issue, to me it happened after trying to install the CH-AC plugin,
I even tried to factory reset

**Comment by DadzBoyz** [comment]
Same problem here.
Hoobs v. 4.0.100
Installed on Raspberry Pi 4 hardware
Raspberry Pi plugged into switch plugged directly into Router
Hoobs, iPhone, other Apple Home devices are all on the same IP range and Subnet
No restrictions on router (Netgear Orbi with one Satellite)
Hoobs is not listed in "Hubs and Bridges" in Home app settings
Change name of hub in router settings
Cleared Cache for each installed bridge
Removed each bridge
Reinstalled each bridge

Try to install any bridge:
Apple Home scans QR code for bridge
Tap on "Add to Home"
Home says "Connecting to Bridge, This may take a few minutes"
Switches to say "Make sure your router is powered on and connected with ethernet cable"
Pinwheel spins.....
"Accessory Not Found, Make sure accessory is powered on and connected to your router with an ethernet cable, then try again."

H E L P....  please.

**Comment by mkellsy** [comment]
The Netgear Orbi is a unique beast. If your phone is connected to the satellite and your HOOBS Box is connected to the main AP, bonjour won't work.

Options
1. You can disable the satellite and pair the bridge. Then re enable the satellite
2. Connect your HOOBS Box with a wire and then add an Apple TV with a wire to the same switch.
3. Try to change the advertiser on the bridge to Caio. Bridge Screen -> [Select Bridge] -> In the settings there is an advertiser field, select Caio.

**Comment by xkyle22x** [comment]
> Same problem here.
> 
> Hoobs v. 4.0.100
> 
> Installed on Raspberry Pi 4 hardware
> 
> Raspberry Pi plugged into switch plugged directly into Router
> 
> Hoobs, iPhone, other Apple Home devices are all on the same IP range and Subnet
> 
> No restrictions on router (Netgear Orbi with one Satellite)
> 
> Hoobs is not listed in "Hubs and Bridges" in Home app settings
> 
> Change name of hub in router settings
> 
> Cleared Cache for each installed bridge
> 
> Removed each bridge
> 
> Reinstalled each bridge
> 
> 
> 
> Try to install any bridge:
> 
> Apple Home scans QR code for bridge
> 
> Tap on "Add to Home"
> 
> Home says "Connecting to Bridge, This may take a few minutes"
> 
> Switches to say "Make sure your router is powered on and connected with ethernet cable"
> 
> Pinwheel spins.....
> 
> "Accessory Not Found, Make sure accessory is powered on and connected to your router with an ethernet cable, then try again."
> 
> 
> 
> H E L P....  please.

So I just had this same issue yesterday, the fix for me was first power cycle/reboot your router and your home hubs, then go into the UI bridge page that u want to add and change the name of the bridge and click the generate new identifier button and click save down the bottom. 
Then up the top click "clear cache".
Then try to add again to home app.

**Comment by xkyle22x** [comment]
> I am experiencing this, as well, step for step. I’ve tried all of the advise listed in this thread, to no avail.
> 
> 
> 
> I have no idea as to what to even try next.

Matt are you trying to scan the code or are u trying to add from the home app by clicking add accessories / do not have code can not scan? 

If u haven't already try the latter

**Comment by mkellsy** [comment]
If HOOBS and your iPhone/iPad on the same network?
Are you using a mesh network?
You may have to connect HOOBS to the main AP, and your phone on the main AP. Once it is paired you can move things back.

Another option is to add an Apple TV or Home Pod to your network. These act as relays.

**Comment by matt-chadbourne-sc** [comment]
Could it be possible that I have a damaged or defective HOOBS device?

**Comment by mkellsy** [comment]
I don't think so. If you do send an email to info@hoobs.com for further troubleshooting. We don't address RMAs on GitHub.

The reason why I am asking you to rename the bridge is the name generates the UUID and HomeKit may be holding on to it. When you install a plugin you are asked to create a bridge. This defaults to "<Plugin Name> Bridge". You can change this to anything, maybe "<Plugin Name> Bridge 2". This will generate a new UUID and it may me able to be added to HomeKit.

**Comment by xkyle22x** [comment]
> @mkellsy Just tried renaming each bridge again, as well as generating a new identifier for each bridge. Like each time I’ve tried this past week, this didn’t resolve the issue.
> 
> 
> 
> I’ve been talking with a representative via info@hoobs.org, by the name of Eddie. So far, his suggested methods have not worked either.

Matt after renaming the bridge and be identifier did u hit clear cashe??

**Comment by matt-chadbourne-sc** [comment]
@mkellsy I'm using a standard CenturyLink C4000BG

**Comment by raideepsingh** [comment]
** 😒 **

**Comment by haavem** [comment]
I suspect the router/modem is the issue. I just switched to CenturyLink from Cox with a Netgear to the same one @matt-chadbourne-sc mentioned and suddenly I can't connect either. Did you ever manage to find a fix?

**Comment by divinetomedy** [comment]
Hey folks, I'm a total networking idiot BUT I did something that worked for me. And that is:

CONNECT TO WIFI.

My phone wasn't on Wifi and I was getting the Apple message (in the Apple Home app), "Connecting to Bridge, This may take a few minutes [pause] Make sure your router is powered on and connected with ethernet cable [pause] Accessory Not Found, Make sure accessory is powered on and connected to your router with an ethernet cable, then try again."

And then I noticed I wasn't on wifi, so I turned it on, immediately tried again, and it worked.
